### PR TITLE
Add `Slot` component

### PR DESCRIPTION
**Pull request by jjenzz**
Allows consumers to turn any of our components into a "slot" when wrapping to create more closed APIs. For example:

```jsx
function Tooltip({ label, children }) {
	return (
		<TooltipPrimitive.Root>
			<TooltipPrimitive.Trigger as={Slot}>
				{children}
			</TooltipPrimitive.Trigger>
			<TooltipPrimitive.Content>
				{label}
			</TooltipPrimitive.Content>
		</TooltipPrimitive.Root>
	);
}

// Usage
<Tooltip label="Notifications" sideOffset={5}>
	<button onClick={handleNotificationsClick}>
		<svg><path /></svg>
	</button>
</Tooltip>
```

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code.
- [ ] Add or edit tests to reflect the change (run `yarn test`).
- [X] Add or edit Storybook examples to reflect the change (run `yarn dev`).
- [ ] Add documentation to support any new features.

This pull request:

- [ ] Fixes a bug in an existing package
- [ ] Adds additional features/functionality to an existing package
- [ ] Updates documentation or example code
- [X] Other

### CODE REVIEW

File: packages/react/scroll-area/src/ScrollArea.tsx [hunk 1]
```diff
@@ -249,21 +249,21 @@ type ScrollAreaNoNativeFallbackPrimitive = Polymorphic.ForwardRefComponent<
 >;
 
 const ScrollAreaNoNativeFallback = React.forwardRef(function ScrollArea(props, forwardedRef) {
-  const { children, ...restProps } = {
+  const { ...restProps } = {
```
Comment (benoitgrelard) [inline packages/react/scroll-area/src/ScrollArea.tsx:252]: nit: could just be `restProps = `now right?

**Comment by benoitgrelard** [comment]
Also just needs versioning

File: packages/react/announce/src/Announce.tsx [hunk 2]
```diff
@@ -96,10 +96,10 @@ const Announce = React.forwardRef((props, forwardedRef) => {
   const {
     selector = getSelector(NAME),
     'aria-relevant': ariaRelevant,
-    children,
     type = 'polite',
     role = ROLES[type],
     regionIdentifier,
+    children,
```
[the remark this answers is not in the document]
Comment (jjenzz) [reply packages/react/announce/src/Announce.tsx:102]: Ah yeah I removed it at first then realised it was needed... I'll put it back where it was

**Review by benoitgrelard — APPROVED** [review]

**Comment by benoitgrelard** [comment]
There's still a little bit of weirdness around types vs. js expectations, etc.
Probably not something we need to worry too much but let's catchup in the morning.

**Comment by benoitgrelard** [comment]
Added merging of props (and composing event handlers)

**Review by benoitgrelard — APPROVED** [review]

File: packages/react/scroll-area/src/ScrollArea.tsx [hunk 3]
```diff
@@ -707,7 +704,7 @@ const ScrollAreaViewport = React.forwardRef(function ScrollAreaViewport(props, f
 
 ScrollAreaViewport.displayName = VIEWPORT_NAME;
 
-/* -------------------------------------------------------------------------------------------------
+/* ---------------------------------------------------  ----------------------------------------------
```
Comment (jjenzz) [inline packages/react/scroll-area/src/ScrollArea.tsx:707]: Oops, looks like an accidental space was added in the middle here haha

File: packages/react/slot/src/Slot.tsx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+ * Slottable
+ * -----------------------------------------------------------------------------------------------*/
+
+const Slottable = ({ children }: { children: React.ReactNode }) => {
+  return children as React.ReactElement;
+};
+
+/* ---------------------------------------------------------------------------------------------- */
+
+function mergeProps<P extends { [key: string]: any }>(baseProps: P, otherProps: P) {
```
Comment (jjenzz) [inline packages/react/slot/src/Slot.tsx:71]: I actually just thought, why do we need the generic? It made me think we're trying to infer something here but do we need inferrence? Would this suffice:

```suggestion
type AnyProps = Record<string, any>;
function mergeProps(baseProps: AnyProps, otherProps: AnyProps) {
```

File: packages/react/scroll-area/src/ScrollArea.tsx (hunk 3 above)
Comment (benoitgrelard) [reply packages/react/scroll-area/src/ScrollArea.tsx:707]: 😀

File: packages/react/slot/src/Slot.tsx (hunk 4 above)
Comment (benoitgrelard) [reply packages/react/slot/src/Slot.tsx:71]: Yep, works, bonus we can remove the `as any`

File: packages/react/menu/src/Menu.tsx [hunk 5]
```diff
@@ -306,14 +303,14 @@ const MenuImpl = React.forwardRef((props, forwardedRef) => {
                     }
                   })}
                 >
-                  <RovingFocusGroup
-                    reachable={itemsReachable}
-                    onReachableChange={setItemsReachable}
-                    orientation="vertical"
-                    loop={loop}
-                  >
-                    <MenuContext.Provider value={context}>{children}</MenuContext.Provider>
```
Comment (Nazeh) [inline packages/react/menu/src/Menu.tsx:315]: How can I add `MenuItem` in this case?

File: packages/react/menu/src/Menu.tsx [hunk 6]
```diff
@@ -306,14 +303,14 @@ const MenuImpl = React.forwardRef((props, forwardedRef) => {
...
-                  </RovingFocusGroup>
+                  <MenuContext.Provider value={context}>
+                    <RovingFocusGroup
+                      reachable={itemsReachable}
+                      onReachableChange={setItemsReachable}
+                      orientation="vertical"
+                      loop={loop}
+                    />
+                  </MenuContext.Provider>
```
Comment (benoitgrelard) [inline packages/react/menu/src/Menu.tsx:313]: My bad @jjenzz, I missed that one… 😞

**Pull request merged**